src/main/kotlin/com/cultureamp/eventsourcing/Aggregate.kt

```diff
@@ -125,7 +125,7 @@ interface AggregateWithProjection<UC : UpdateCommand, UE : UpdateEvent, Err : Do
 
 interface AggregateConstructor<CC : CreationCommand, CE : CreationEvent, Err : DomainError, UC : UpdateCommand, UE : UpdateEvent, M : EventMetadata, Self : Aggregate<UC, UE, Err, M, Self>> {
     fun created(event: CE): Self
-    fun create(command: CC, metadata: M): Either<Err, CE>
+    fun create(command: CC, metadata: M): Either<Err, Pair<CE, List<UE>>>
```

Comment:
Is there no need to maintain compatibility?

Reply:
Very well spotted!! I was waiting to see if someone would notice this. We have indeed made a breaking change here. My hope in the future is the decouple these interfaces from the underlying implementation, having it use pure functions and translating these interfaces into those underlying functions. I didn't want to bite that off in this change, and the pragmatist in me thought even though this is technically breaking, Kestrel hasn't really been adopted in many places and it kind of makes sense to have these interfaces be "complete" (as in providing the most capability) as a default.

Maybe that's a bit brash though, do you think it's worth the extra work now to avoid this breaking change?

Reply:
Maybe ill give it another 30min timebox

Reply:
Note that `SimpleAggregate` and `SimpleAggregateConstructor` remain untouched which are sort of "basic" construction mode recommended in the README

Reply:
we wouldn't break our existing code, should be fine here.